### PR TITLE
Fixing accessibility to Aria nodes and input fields

### LINKED ISSUE #4800: [Outreachy Task Submission] Aria Progressbar nodes and  input fields must have an accessible name
## Description 
Add an accessible name for both situations so that screen readers can identify and announce the purpose of the input field to users with disabilities.
Screenshot 2024-03-11 at 11 10 45

### PR DESCRIPTION
**Pull request by MundiaNderi**
# Description

Input fields and ARIA progress nodes do not have accessible names or description. This PR fixes this.

## Test
To test:
- Install [axe chrome extension](https://axe.deque.com/install-success).  Open the developer console tab and scan the page.
- There should be no aria accessible name errors
-  Make a change in one of the files in the PR by removing the ariaLabel. 
- Reload and scan the page - you should see aria accessibility errors.

## Fixes
Fixes: https://github.com/ushahidi/platform/issues/4800#event-12132665123

### CODE REVIEW

**Comment by MundiaNderi** [comment]
@Angamanga this PR is ready for review.

**Comment by Angamanga** [comment]
Thank you for your PR @MundiaNderi  🙏. Could you please add a short description and a checklist for testing the fix as well and then we'll review it 👍

File: apps/web-mzima-client/src/app/shared/components/search-form/search-form.component.scss [hunk 1]
```diff
@@ -687,3 +687,15 @@
...
+.sr-only {
+  position: absolute;
+  width: 1px;
+  height: 1px;
+  padding: 0;
+  margin: -1px;
+  overflow: hidden;
+  clip: rect(0, 0, 0, 0);
+  white-space: nowrap; /* Added for accessibility */
+  border: 0;
```
Comment (Angamanga) [inline apps/web-mzima-client/src/app/shared/components/search-form/search-form.component.scss]: Do you think this could be useful as a general style? If so, I suggest you move the class to styles.scss 👍

**Review by Angamanga — CHANGES_REQUESTED** [review]
Thank you for your contribution! A small question and a change needed.

File: apps/web-mzima-client/src/app/shared/components/search-form/search-form.component.scss (hunk 1 above)
Comment (MundiaNderi) [reply apps/web-mzima-client/src/app/shared/components/search-form/search-form.component.scss]: @Angamanga That is outdated - following this discovery here : https://github.com/ushahidi/platform-client-mzima/pull/830#issuecomment-2008955786. Git pull the recent changes.

File: package-lock.json
Comment (Angamanga) [inline package-lock.json:1]: Remove this file from the PR

File: tsconfig.json
Comment (Angamanga) [inline tsconfig.json:1]: Is this change needed for this PR?

**Review by Angamanga — CHANGES_REQUESTED** [review]
Nearly there, just remove the package-lock and the new tsconfig file 👍

File: tsconfig.json
Comment (MundiaNderi) [reply tsconfig.json:1]: Resolved tsconfig. It got in there when I was testing Cypress for accessibility testing. I'll go ahead and fix the rest.

**Comment by MundiaNderi** [comment]
@Angamanga it's now resolved.

**Review by Angamanga — APPROVED** [review]
Approved 🎉 Thank you for your contribution!
